frontera/contrib/backends/redis_backend/__init__.py

```diff
@@ -0,0 +1,443 @@
+                break
+
+
+class RedisPipeline(object):
+    def __init__(self, pool):
+        connection = StrictRedis(connection_pool=pool)
+        self._pipeline = connection.pipeline()
+        self._logger = logging.getLogger("redis_backend.RedisPipeline")
+
+    def __getattr__(self, _api):
+        return getattr(self._pipeline, _api)
+
+    def execute(self):
+        timeout = _get_retry_timeouts()
+        stack = self._pipeline.command_stack
+        while True:
+            try:
+                return self._pipeline.execute()
+            except ConnectionError:
+                self._logger.exception("Connection to Redis failed when executing pipeline")
+                pause = timeout.next()
+                if pause is None:
+                    break
+                sleep(pause)
+                self._pipeline.command_stack = stack
+            except ResponseError:
+                self._logger.exception("Redis operation failed when executing pipeline")
+                break
+
+
+class RedisQueue(Queue):
+    MAX_SCORE = 1.0
+    MIN_SCORE = 0.0
+    SCORE_STEP = 0.01
+
+    def __init__(self, manager, pool, partitions, delete_all_keys=False):
+        settings = manager.settings
+        codec_path = settings.get('REDIS_BACKEND_CODEC')
+        encoder_cls = load_object(codec_path + ".Encoder")
+        decoder_cls = load_object(codec_path + ".Decoder")
+        self._encoder = encoder_cls(manager.request_model)
+        self._decoder = decoder_cls(manager.request_model, manager.response_model)
+        self._redis = RedisOperation(pool)
+        self._redis_pipeline = RedisPipeline(pool)
+        self._partitions = [i for i in range(0, partitions)]
+        self._partitioner = Crc32NamePartitioner(self._partitions)
+        self._logger = logging.getLogger("redis_backend.queue")
+
+        if delete_all_keys:
+            self._redis.flushdb()
+
+    def _get_items(self, partition_id, start, now_ts, queue, max_requests_per_host, max_host_items, count,
+                   max_n_requests, to_remove):
+        for data in self._redis.zrevrange(partition_id, start=start, end=max_n_requests + start):
+            start += 1
+            item = unpackb(data, use_list=False)
+            timestamp, fprint, host_crc32, _, score = item
+            if timestamp > now_ts:
+                continue
+            if host_crc32 not in queue:
+                queue[host_crc32] = []
+            if max_requests_per_host is not None and len(queue[host_crc32]) > max_requests_per_host:
+                continue
+            queue[host_crc32].append(item)
+            if len(queue[host_crc32]) > max_host_items:
+                max_host_items = len(queue[host_crc32])
+            count += 1
+            to_remove.append(data)
+            if count >= max_n_requests:
+                break
+        return start, count, max_host_items
+
+    def get_next_requests(self, max_n_requests, partition_id, **kwargs):
```

Comment:
As I see the ``min_hosts`` isn't respected. would it be possible to make support it?